homework/index.js

```diff
@@ -30,18 +36,63 @@
     return elem;
   }
 
+  
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
+    fetchJSON(url)
+      .then(repositoryData => {
+        const select = createAndAppend('select', root);
+        createAndAppend('option', select, { text: 'Click here to choose a Repository' });
```

Comment:
I'm still missing:

1. Data is loaded for the first repository when the app/page loads
2. The repos are sorted by name

Reply:
Sorted by name, still thinking how to start the first repo onload

Reply:
Sorting looks good! Let me know if you need any pointers for first repo onload.

Reply:
yes, please

Reply:
What you need to do is basically the following:

1. Move the function that you send in as a callback to the eventhandler on line 53 to a separate function
2. Call this function in the event-handler on line 53
3. Call this function when the page loads

Is it clearer how to proceed?